src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,353 @@
+            if (currentSearchDistance < getMaxVision())
+            {
+                currentSearchDistance += START_SEARCH_DISTANCE;
+            }
+            else
+            {
+                currentSearchDistance = START_SEARCH_DISTANCE;
+                return AIState.GUARD_PATROL;
+            }
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        return AIState.GUARD_GET_TARGET;
+    }
+
+    private AxisAlignedBB getTargetableArea(double range)
+    {
+        return this.worker.getEntityBoundingBox().expand(range, HEIGHT_DETECTION_RANGE, range);
+    }
+
+    /**
+     * Getter for the vision or attack distance.
+     *
+     * @return the max vision.
+     */
+    private double getMaxVision()
+    {
+        BuildingGuardTower guardTower = (BuildingGuardTower) worker.getWorkBuilding();
+        return (guardTower == null) ? 0 : (MAX_ATTACK_DISTANCE + guardTower.getBonusVision());
+    }
+
+    /**
+     * Getter calculating how many arrows the guard may shoot or deal sword hits until restock.
+     *
+     * @return the amount.
+     */
+    protected int getMaxAttacksUntilRestock()
+    {
+        BuildingGuardTower guardTower = (BuildingGuardTower) worker.getWorkBuilding();
+        return (guardTower == null) ? 0 : (MAX_ATTACKS + guardTower.getBuildingLevel());
+    }
+
+    /**
+     * Lets the guard patrol inside the colony area searching for mobs.
+     *
+     * @return the next state to go.
+     */
+    protected AIState patrol()
+    {
+        worker.setAIMoveSpeed(1);
+
+        if (currentPatrolTarget == null)
+        {
+            currentPatrolTarget = getRandomBuilding();
+        }
+
+        if (worker.isWorkerAtSiteWithMove(currentPatrolTarget, PATH_CLOSE))
+        {
+            currentPatrolTarget = null;
+        }
+
+        return AIState.GUARD_SEARCH_TARGET;
+    }
+
+    /**
+     * Gets a random building from his colony.
+     *
+     * @return a random blockPos.
+     */
+    private BlockPos getRandomBuilding()
+    {
+        if (worker.getColony() == null || getOwnBuilding() == null)
+        {
+            return worker.getPosition();
+        }
+
+        final Collection<AbstractBuilding> buildingList = worker.getColony().getBuildings().values();
+        final Object[] buildingArray = buildingList.toArray();
+
+        final int random = worker.getRandom().nextInt(buildingArray.length);
+        final AbstractBuilding building = (AbstractBuilding) buildingArray[random];
+
+        if (building instanceof BuildingGuardTower
+                || BlockPosUtil.getDistance2D(building.getLocation(), this.getOwnBuilding().getLocation()) > getPatrolDistance())
+        {
+            return this.getOwnBuilding().getLocation();
+        }
+
+        return building.getLocation();
+    }
+
+    /**
+     * Called when a guard killed an entity.
+     *
+     * @param killedEntity the entity being killed.
+     */
+    protected void onKilledEntity(final EntityLivingBase killedEntity)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "killedEntity". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)